### ISSUE TITLE
Java API doesn't pass error details. Error Code is 0 (invalid) when there is no SQL State

### ISSUE THREAD

**Issue by wparuch** [flyway/flyway#3145]
###### Which version and edition of Flyway are you using?
7.7.3 latest

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Java API

###### Which database are you using? (Type & version)
DB2 11.5.5

###### Which operating system are you using?
MacOs / Windows

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)
Run migration that contains:
`CALL SYSPROC.ADMIN_MOVE_TABLE(current_schema, 'SOMETABLE', 'SOMETABLE3', '', 'MOVE')`. `SOMETABLE` should exist, `SOMETABLE3` should not be there.

###### What did you expect to see?
Exception with `Error Code` filled, like:
```
org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: 
Migration V1.11__blabla.sql failed
----------------------------------------------------------------------
SQL State  : null
Error Code : -2211
Location   : migrations/db2/V1.11__blabla.sql (/some/dir/migrations/db2/V1.11__blabla.sql)
Line       : 18
Statement  : CALL SYSPROC.ADMIN_MOVE_TABLE(current_schema, 'SOMETABLE', 'SOMETABLE3', '', 'MOVE')
```

###### What did you see instead?
```
org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: 
Migration V1.11__blabla.sql failed
----------------------------------------------------------------------
SQL State  : null
Error Code : 0
Location   : migrations/db2/V1.11__blabla.sql (/some/dir/migrations/db2/V1.11__blabla.sql)
Line       : 18
Statement  : CALL SYSPROC.ADMIN_MOVE_TABLE(current_schema, 'SOMETABLE', 'SOMETABLE3', '', 'MOVE')

	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:427)
	at org.flywaydb.core.internal.command.DbMigrate.access$200(DbMigrate.java:56)
	at org.flywaydb.core.internal.command.DbMigrate$3.call(DbMigrate.java:331)
	at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:66)
	at org.flywaydb.core.internal.command.DbMigrate.applyMigrations(DbMigrate.java:328)
	at org.flywaydb.core.internal.command.DbMigrate.migrateGroup(DbMigrate.java:291)
	at org.flywaydb.core.internal.command.DbMigrate.access$100(DbMigrate.java:56)
	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:195)
	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:192)
	at org.flywaydb.core.internal.jdbc.TableLockingExecutionTemplate$1.call(TableLockingExecutionTemplate.java:38)
	at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:66)
	at org.flywaydb.core.internal.jdbc.TableLockingExecutionTemplate.execute(TableLockingExecutionTemplate.java:33)
	at org.flywaydb.core.internal.database.base.Connection.lock(Connection.java:103)
	at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:141)
	at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:192)
	at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:152)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:216)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:165)
	at org.flywaydb.core.Flyway.execute(Flyway.java:572)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:165)
    ...
Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: 
Migration V1.11__blabla.sql failed
----------------------------------------------------------------------
SQL State  : null
Error Code : 0
Location   : migrations/db2/V1.11__blabla.sql (/some/dir/migrations/db2/V1.11__blabla.sql)
Line       : 18
Statement  : CALL SYSPROC.ADMIN_MOVE_TABLE(current_schema, 'SOMETABLE', 'SOMETABLE3', '', 'MOVE')

	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.handleException(DefaultSqlScriptExecutor.java:271)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:218)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.execute(DefaultSqlScriptExecutor.java:126)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.executeOnce(SqlMigrationExecutor.java:78)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.lambda$execute$0(SqlMigrationExecutor.java:67)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor$$Lambda$751/0x0000000000000000.call(Unknown Source)
	at org.flywaydb.core.internal.database.DefaultExecutionStrategy.execute(DefaultExecutionStrategy.java:27)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.execute(SqlMigrationExecutor.java:66)
	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:424)
	... 105 common frames omitted
Caused by: com.ibm.db2.jcc.am.SqlException: DB2 SQL Error: SQLCODE=-2211, SQLSTATE=     , SQLERRMC=11;, DRIVER=4.26.14
	at com.ibm.db2.jcc.am.b7.a(b7.java:815)
	at com.ibm.db2.jcc.am.b7.a(b7.java:66)
	at com.ibm.db2.jcc.am.b7.a(b7.java:140)
	at com.ibm.db2.jcc.am.k4.c(k4.java:2824)
	at com.ibm.db2.jcc.t4.ab.x(ab.java:1827)
	at com.ibm.db2.jcc.t4.ab.n(ab.java:950)
	at com.ibm.db2.jcc.t4.ab.f(ab.java:156)
	at com.ibm.db2.jcc.t4.p.e(p.java:81)
	at com.ibm.db2.jcc.t4.av.k(av.java:175)
	at com.ibm.db2.jcc.am.k4.ao(k4.java:2413)
	at com.ibm.db2.jcc.am.k4.a(k4.java:3383)
	at com.ibm.db2.jcc.am.k4.e(k4.java:1131)
	at com.ibm.db2.jcc.am.k4.execute(k4.java:1110)
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:95)
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
	at sun.reflect.GeneratedMethodAccessor65.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.proxy.$Proxy99.execute(Unknown Source)
	...
	at org.flywaydb.core.internal.jdbc.JdbcTemplate.executeStatement(JdbcTemplate.java:241)
	at org.flywaydb.core.internal.sqlscript.ParsedSqlStatement.execute(ParsedSqlStatement.java:111)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:206)
	... 112 common frames omitted
```

As you can see SQL Exception does contain details:
`DB2 SQL Error: SQLCODE=-2211, SQLSTATE=     , SQLERRMC=11;, DRIVER=4.26.14`
Please pass them in `FlywayMigrateException`.

**Comment by DoodleBobBuffPants** [comment]
I'm unable to reproduce this. Could you provide exact steps to reproduce this?

**Comment by wparuch** [comment]
Sure. Actually I reproduced it with even simpler scenario on 7.8.0:

Migration file with missing closing bracket:
```
CREATE TABLE "SOMETABLE" (
    "ID" CHAR(36) NOT NULL,
    "VALUE" CLOB(10240);
```

Exception:
```
org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: 
Migration V5.0__create_test_table.sql failed
---------------------------------------------
SQL State  : null
Error Code : 0
Location   : migrations/db2/V5.0__create_test_table.sql (/some/dir/migrations/db2/V5.0__create_test_table.sql)
Line       : 1
Statement  : CREATE TABLE "SOMETABLE" (
    "ID" CHAR(36) NOT NULL,
    "VALUE" CLOB(10240);

	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:357)
	at org.flywaydb.core.internal.command.DbMigrate.lambda$applyMigrations$1(DbMigrate.java:262)
	at org.flywaydb.core.internal.command.DbMigrate$$Lambda$572/0x0000000000000000.call(Unknown Source)
	at org.flywaydb.core.internal.jdbc.Transac§tionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:66)
	at org.flywaydb.core.internal.command.DbMigrate.applyMigrations(DbMigrate.java:261)
	at org.flywaydb.core.internal.command.DbMigrate.migrateGroup(DbMigrate.java:234)
	at org.flywaydb.core.internal.command.DbMigrate.lambda$migrateAll$0(DbMigrate.java:140)
	at org.flywaydb.core.internal.command.DbMigrate$$Lambda$570/0x0000000000000000.call(Unknown Source)
	at org.flywaydb.core.internal.jdbc.TableLockingExecutionTemplate$1.call(TableLockingExecutionTemplate.java:38)
	at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:66)
	at org.flywaydb.core.internal.jdbc.TableLockingExecutionTemplate.execute(TableLockingExecutionTemplate.java:33)
	at org.flywaydb.core.internal.database.base.Connection.lock(Connection.java:103)
	at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:141)
	at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:140)
	at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:100)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:216)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:165)
	at org.flywaydb.core.Flyway.execute(Flyway.java:571)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:165)
	...
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:389)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:333)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:157)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:415)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1791)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:594)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:516)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:324)
	at org.springframework.beans.factory.support.AbstractBeanFactory$$Lambda$303/0x0000000000000000.getObject(Unknown Source)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:897)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:879)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:551)
	...
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:686)
    ...
Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: 
Migration V5.0__create_test_table.sql failed
---------------------------------------------
SQL State  : null
Error Code : 0
Location   : migrations/db2/V5.0__create_test_table.sql (/some/dir/migrations/db2/V5.0__create_test_table.sql)
Line       : 1
Statement  : CREATE TABLE "SOMETABLE" (
    "ID" CHAR(36) NOT NULL,
    "VALUE" CLOB(10240);

	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.handleException(DefaultSqlScriptExecutor.java:275)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:218)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.execute(DefaultSqlScriptExecutor.java:126)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.executeOnce(SqlMigrationExecutor.java:78)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.lambda$execute$0(SqlMigrationExecutor.java:67)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor$$Lambda$751/0x0000000000000000.call(Unknown Source)
	at org.flywaydb.core.internal.database.DefaultExecutionStrategy.execute(DefaultExecutionStrategy.java:27)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.execute(SqlMigrationExecutor.java:66)
	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:354)
	... 124 common frames omitted
Caused by: com.ibm.db2.jcc.am.SqlSyntaxErrorException: DB2 SQL Error: SQLCODE=-104, SQLSTATE=42601, SQLERRMC=END-OF-STATEMENT;"VALUE" CLOB(10240);), DRIVER=4.26.14
	at com.ibm.db2.jcc.am.b7.a(b7.java:810)
	at com.ibm.db2.jcc.am.b7.a(b7.java:66)
	at com.ibm.db2.jcc.am.b7.a(b7.java:140)
	at com.ibm.db2.jcc.am.k4.c(k4.java:2824)
	at com.ibm.db2.jcc.am.k4.d(k4.java:2808)
	at com.ibm.db2.jcc.am.k4.b(k4.java:2168)
	at com.ibm.db2.jcc.t4.ab.k(ab.java:446)
	at com.ibm.db2.jcc.t4.ab.c(ab.java:104)
	at com.ibm.db2.jcc.t4.p.b(p.java:38)
	at com.ibm.db2.jcc.t4.av.h(av.java:124)
	at com.ibm.db2.jcc.am.k4.ak(k4.java:2163)
	at com.ibm.db2.jcc.am.k4.a(k4.java:3366)
	at com.ibm.db2.jcc.am.k4.e(k4.java:1131)
	at com.ibm.db2.jcc.am.k4.execute(k4.java:1110)
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:95)
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	...
	at org.flywaydb.core.internal.jdbc.JdbcTemplate.executeStatement(JdbcTemplate.java:241)
	at org.flywaydb.core.internal.sqlscript.ParsedSqlStatement.execute(ParsedSqlStatement.java:111)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:206)
	... 131 common frames omitted
```

**Comment by DoodleBobBuffPants** [comment]
I'm still unable to reproduce this - all error codes are appearing as expecting

Could you try using [this](https://mvnrepository.com/artifact/com.ibm.db2/jcc/11.5.0.0) driver

If the above fails, can you confirm which driver you are using and provide a minimal reproduction project?

**Comment by wparuch** [comment]
@DoodleBobBuffPants hmm. I use JDBC 4.0 Driver, version 4.26.14 (db2jcc4.jar, dedicated for DB2 11.5), avaliable here:
https://www.ibm.com/support/pages/db2-jdbc-driver-versions-and-downloads
This problem occurs on both DB2 11.5.1 and DB2 11.5.5 which I use.

I'll check it with your driver.

**Comment by wparuch** [comment]
Yep, result is exactly the same, as this is the same driver ;)

**Comment by DoodleBobBuffPants** [comment]
Closing until we can get a reproduction

**Comment by wparuch** [comment]
It's creative accounting. Problem is there, yet I have no resources to create new project with all configs I use, spring boot, hibernate and so on. I'll post it there as soon as I'll find some spare time.

**Comment by wparuch** [comment]
I narrowed down the problem. When I use:
`Flyway.configure().dataSource(dataSource)...` no exception details are provided, but when I manually enter connection parameters:
`Flyway.configure().dataSource("jdbc:db2://localhost:50000/DBNAME", "user", "pass").schemas("schema")`
it works as expected. Yet, this workaround is not an option for me.

I've tried to recreate this with simple `DB2SimpleDataSource`, but this DataSource works as expected.
